MeshLib/MeshGenerators/MeshGenerator.cpp

```diff
@@ -113,6 +114,35 @@ std::vector<MeshLib::Node*> MeshGenerator::generateRegularNodes(
     return nodes;
 }
 
+namespace MeshGenerator
+{
+std::vector<MeshLib::Node*> generateRegularPyramidTopNodes(
+    std::vector<double> const& x_coords,
+    std::vector<double> const& y_coords,
+    std::vector<double> const& z_coords,
+    const MathLib::Point3d& origin)
+{
+    std::vector<Node*> nodes;
+    nodes.reserve((x_coords.size() - 1) * (y_coords.size() - 1) *
+                  (z_coords.size() - 1));
+
+    for (std::size_t i = 0; i < z_coords.size() - 1; i++)
```

Comment:
minor: storing the sizes once is usually faster than asking the vector for its size on every iteration.

Reply:
Maybe the compiler can optimize it because the vector is const.

Reply:
No. It is const in this function, but might not be const elsewhere.

Reply:
Yes, you are right.